app/services/candidate_interface/editable_section.rb

```diff
@@ -0,0 +1,51 @@
+module CandidateInterface
+  class EditableSection
+    include ActiveModel::Model
+    attr_accessor :current_application, :controller_path, :action_name, :params
+
+    Section = Struct.new(:controller, :condition, keyword_init: true)
+
+    def self.all
+      [
+        Section.new(controller: 'CandidateInterface::PersonalDetails'),
+        Section.new(controller: 'CandidateInterface::ContactDetails'),
+        Section.new(controller: 'CandidateInterface::TrainingWithADisability'),
+        Section.new(controller: 'CandidateInterface::InterviewAvailability'),
+        Section.new(controller: 'CandidateInterface::EqualityAndDiversity'),
+        Section.new(controller: 'CandidateInterface::PersonalStatement'),
+        Section.new(controller: 'CandidateInterface::Gcse', condition: :science_gcse?),
+      ]
+    end
+
+    def can_edit?
+      any_offer_accepted? || all_applications_unsubmitted? || editable_section?
```

Comment:
Is `all_applications_unsubmitted?` the only valid condition related to statuses? What if you have a withdrawn application for example?